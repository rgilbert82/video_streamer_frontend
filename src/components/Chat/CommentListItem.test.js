import React               from 'react';
import { shallow }         from 'enzyme';
import { CommentListItem } from '.';

describe(CommentListItem, () => {
  const message = {
    user: {
      id: '123',
      username: 'Alice',
      image_url: 'youtube.com'
    },
    comment: {
      message: 'hello',
      published_at: '2018'
    }
  }
  let component;

  beforeEach(() => {
    component = shallow(<CommentListItem message={message} displayMessage={jest.fn()}/>);
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });
});
