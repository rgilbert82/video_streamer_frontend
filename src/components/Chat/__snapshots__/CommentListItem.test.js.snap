// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class CommentListItem extends _react2.default.Component {
  constructor(props) {
    super(props);

    this.state = {
      user: this.props.message.user,
      comment: this.props.message.comment };

  }

  render() {
    const userPath = \`/users/\${this.state.user.id}\`;
    return (
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this },
        _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this },
          _react2.default.createElement(_reactRouterDom.Link, { to: userPath, __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this },
            _react2.default.createElement('img', { src: this.state.user.image_url, alt: this.state.user.id, __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this }),
            _react2.default.createElement('h4', { __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }, this.state.user.username))),


        _react2.default.createElement('p', { __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this }, this.state.comment.message),
        _react2.default.createElement('small', { __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this }, this.state.comment.published_at)));


  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentListItem
    displayMessage={[Function]}
    message={
        Object {
            "comment": Object {
              "message": "hello",
              "published_at": "2018",
            },
            "user": Object {
              "id": "123",
              "image_url": "youtube.com",
              "username": "Alice",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Link
                    replace={false}
                    to="/users/123"
          >
                    <img
                              alt="123"
                              src="youtube.com"
                    />
                    <h4>
                              Alice
                    </h4>
          </Link>
</div>,
        <p>
          hello
</p>,
        <small>
          2018
</small>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/users/123"
>
            <img
                        alt="123"
                        src="youtube.com"
            />
            <h4>
                        Alice
            </h4>
</Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <img
                alt="123"
                src="youtube.com"
/>,
              <h4>
                Alice
</h4>,
            ],
            "replace": false,
            "to": "/users/123",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "123",
                "src": "youtube.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Alice",
              },
              "ref": null,
              "rendered": "Alice",
              "type": "h4",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hello",
        },
        "ref": null,
        "rendered": "hello",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018",
        },
        "ref": null,
        "rendered": "2018",
        "type": "small",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Link
                        replace={false}
                        to="/users/123"
            >
                        <img
                                    alt="123"
                                    src="youtube.com"
                        />
                        <h4>
                                    Alice
                        </h4>
            </Link>
</div>,
          <p>
            hello
</p>,
          <small>
            2018
</small>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/users/123"
>
              <img
                            alt="123"
                            src="youtube.com"
              />
              <h4>
                            Alice
              </h4>
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt="123"
                  src="youtube.com"
/>,
                <h4>
                  Alice
</h4>,
              ],
              "replace": false,
              "to": "/users/123",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "123",
                  "src": "youtube.com",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Alice",
                },
                "ref": null,
                "rendered": "Alice",
                "type": "h4",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hello",
          },
          "ref": null,
          "rendered": "hello",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018",
          },
          "ref": null,
          "rendered": "2018",
          "type": "small",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
